client/src/main/java/io/pravega/client/batch/BatchClient.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.batch;
+
+import com.google.common.annotations.Beta;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.ReaderGroup;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.Stream;
+import java.util.Date;
+import java.util.Iterator;
+
+/**
+ * Please note this is an experimental API.
+ * 
+ * Used to get metadata about and read from an existing streams.
+ * <p>
+ * All events written to a stream will be visible atomically and their existence reflected in {@link SegmentInfo#getLength()} and 
```

Comment:
I wouldn't use the term "atomic" here, it is going to confuse people. It is better to use more words and explain what we are doing. We are actually making available all segments of the stream available for reading in parallel. Making them available for reading in parallel implies that the order of events in the stream is no longer preserved, which is typically fine for batch jobs. If order is important, then please use the regular event read API.